test/IPNFTMintHelper.sol

```diff
@@ -32,9 +34,11 @@ abstract contract IPNFTMintHelper is Test {
     }
 
     function mintAToken(IReservable ipnft, address to) internal returns (uint256) {
-        uint256 reservationId = reserveAToken(ipnft, to);
+        dealMintpass(to);
         vm.startPrank(to);
-        ipnft.mintReservation(to, reservationId, reservationId, arUri);
+        uint256 reservationId = ipnft.reserve();
+
+        ipnft.mintReservation{value: MINTING_FEE}(to, reservationId, reservationId, arUri);
```

Comment:
this feels weird here to name the mintpassId reservationId as well. Not important in the context of the test, but weird when just reading through the code.

Reply:
well, it's been like that from the very beginnings and as long as there's only 1 user using it (which it is in case of tests), I think, that's obvious. I a previous version I think we even hardcoded a "1" here which oc breaks in more complex scenarios. If there ever is someone going to extend the test cases, they'll stumble upon it and might decide to change it, but imo it's totally fine like that.

Reply:
yes, both are not ideal and I agree that there is redundant logic. For now, I guess it's fine. It just looks weird